### DISCUSSION TITLE
no response received expected atleast 2 bytes

### DISCUSSION THREAD

**Discussion by sachinmishra371** [pymodbus-dev/pymodbus#1045] [Help] [answered]
i struck in rtu485modscan  from a long time,and i am not understanding what to do now,please help.......
<img width="487" alt="modscanscreenshotcrcerror" src="https://user-images.githubusercontent.com/64963567/185293320-b6beee1e-14d1-4ea8-8ec1-31a0402e0c6a.png">

<img width="487" alt="modscanscreenshotconfig" src="https://user-images.githubusercontent.com/64963567/185299461-ac850170-c337-4897-8827-341a1c2d423c.png">

<img width="700" alt="Screenshot 2022-08-17 162759" src="https://user-images.githubusercontent.com/64963567/185290208-2511b8ee-9199-4bd6-a77f-be9072856c8b.png">
<img width="694" alt="modscanscreenshot" src="https://user-images.githubusercontent.com/64963567/185290969-e316c9b4-795a-4002-b789-7faeccb7a462.png">

**Comment by swamper123** [comment, marked as answer]
In your picture you show us the Connection Details for COM6, but in your code it's port COM5.
Might that be it?

**Reply by sachinmishra371** [reply]
> In your picture you show us the Connection Details for COM6, but in your code it's port COM5. Might that be it?

both are on different computer ,like modcsan using com6 to send data using rs485 and another side of rtu cable connected with com5 of different computer so my code taking data from com5.
in .Net language it works but in python using pymodbus it not working.Any sugguestion?

**Reply by janiversen** [reply]
that will not work, modscan as a client expecting to connect to a device, so does your pymodbus setup. 

You need one end to work as a client requesting data from the other end that needs to be a server which will respond to the request.

**Reply by sachinmishra371** [reply]
> that will not work, modscan as a client expecting to connect to a device, so does your pymodbus setup.
> 
> You need one end to work as a client requesting data from the other end that needs to be a server which will respond to the request.

sir,any code  i am fresher here

**Reply by janiversen** [reply]
look at the examples and read the modbus standard.

**Reply by sachinmishra371** [reply]
> look at the examples and read the modbus standard.

i searched too much but i didn't get solution.can i get link

**Reply by janiversen** [reply]
btw. what you are trying to do are very unusual, normally pymodbus is used to “talk” with a physical modbus device. There are examples of both client and server code.

We do what you try to do but within the same computer and only for test purposes see test_example.py

**Reply by janiversen** [reply]
> > look at the examples and read the modbus standard.
> 
> i searched too much but i didn't get solution.can i get link

Just look at pymodbus which you have already found.

**Reply by janiversen** [reply]
You should have all the information you need now.

**Reply by sachinmishra371** [reply]
not working still sir

**Reply by janiversen** [reply]
Well you are the programmer, solve it!

I have no idea why you want to make such a strange configuration, but I am sure you have good reasons...however you cannot expect us to solve your programming issues.

Solely using the examples in the repo, you are able to make a setup with 2 computers, one acting as server and the other as client, so I do not believe we can help further.
